common/src/main/java/org/apache/celeborn/common/network/util/NettyUtils.java

```diff
@@ -182,34 +173,37 @@ public static ByteBufAllocator getByteBufAllocator(
 
   public static ByteBufAllocator getByteBufAllocator(
       TransportConf conf, AbstractSource source, boolean allowCache, int coreNum) {
+    ByteBufAllocator allocator;
     if (conf.getCelebornConf().networkShareMemoryAllocator()) {
-      return getSharedByteBufAllocator(
-          conf.getCelebornConf(),
-          source,
-          allowCache && conf.getCelebornConf().networkMemoryAllocatorAllowCache());
-    }
-    int arenas;
-    if (coreNum != 0) {
-      arenas = coreNum;
+      allocator =
+          getSharedByteBufAllocator(
+              conf.getCelebornConf(),
+              allowCache && conf.getCelebornConf().networkMemoryAllocatorAllowCache());
     } else {
-      arenas = conf.getCelebornConf().networkAllocatorArenas();
-    }
-    ByteBufAllocator allocator =
-        createByteBufAllocator(
-            conf.getCelebornConf().networkMemoryAllocatorPooled(),
-            conf.preferDirectBufs(),
-            allowCache,
-            arenas);
-    if (conf.getCelebornConf().networkMemoryAllocatorPooled()) {
-      pooledByteBufAllocators.add((PooledByteBufAllocator) allocator);
+      int arenas;
+      if (coreNum != 0) {
+        arenas = coreNum;
+      } else {
+        arenas = conf.getCelebornConf().networkAllocatorArenas();
+      }
+      allocator =
+          createByteBufAllocator(
+              conf.getCelebornConf().networkMemoryAllocatorPooled(),
+              conf.preferDirectBufs(),
+              allowCache,
+              arenas);
+      if (conf.getCelebornConf().networkMemoryAllocatorPooled()) {
+        pooledByteBufAllocators.add((PooledByteBufAllocator) allocator);
+      }
     }
     if (source != null) {
-      String poolName = "default-netty-pool";
+      String poolName;
       Map<String, String> labels = new HashMap<>();
-      String moduleName = conf.getModuleName();
-      if (!moduleName.isEmpty()) {
-        poolName = moduleName;
-        int index = allocatorsIndex.compute(moduleName, (k, v) -> v == null ? 0 : v + 1);
+      if (conf.getCelebornConf().networkMemoryAllocatorAllowCache()) {
+        poolName = allowCache ? "celeborn-netty-shared-cache-pool" : "celeborn-netty-shared-pool";
```

Comment:
addressed https://github.com/apache/celeborn/pull/3007#discussion_r1893437934

Reply:
Seems if `networkMemoryAllocatorAllowCache` is true, the grafana dashboard would also be impacted.